Power generator fuel refactor

Aranclanos opened the pull request:
### Purpose
Adds a player error message for when the generator already contains a sheet if a player tries to add one.

### EDIT:
Refactors fuel consumption by the power generators.

Removes SolidPlasma.cs, all fuel consumption will now be handled by PowerGenerator.cs
Fixes power generators burning plasma instantly after the first sheet of the stack gets removed.
Players will now be allowed to add stacks to the generator.
Adds a player message warning for when the player tries to turn on a generator without power.
A plasma sheet will no longer be required for maps, the bool startAsOn will add a the corresponding fuel of one sheet to the generator at the start.

Bod9001 committed:
Fixing something in cargo,This is fixes reactions,

Adds editor preview animation
Fixes some errors

Bod9001 committed:
Merge branch 'develop' into Cargo_fixed

Aranclanos committed:
Adds a player error message for when the generator already contains a sheet if a player tries to add one.

ReiderJack committed:
Merge pull request #5033 from Bod9001/Cargo_fixed

Fixing something in cargo,This is fixes reactions,

NoooneyDude commented on UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:
```diff
@@ -154,8 +154,7 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 		if (interaction.TargetObject != gameObject) return false;
-		if (interaction.HandObject != null &&
-				!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma)) return false;
+		if (!Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma)) return false;
```
Won't this mean you will not be able to toggle the power state of the generator?

NoooneyDude commented on UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:
```diff
@@ -164,6 +163,11 @@ public void ServerPerformInteraction(HandApply interaction)
+			{
+				Chat.AddWarningMsgFromServer(interaction.Performer, "The generator already contains a sheet!");
+				return;
+			}
```
The generator uses ItemStorage now, so it should be able to contain up to fifty sheets.

corp-0 replied:
it doesn't stack for some reason

Aranclanos replied:
ye im making it stack ETA a couple hours

Aranclanos committed:
Fixes power generators not responding to non-plasma sheet related clicks

NoooneyDude committed:
Canister touchup

Bod9001 committed:
Adds test for Objects that get set at 0,0  And removed some stuff that was causing warnings

Bod9001 committed:
oh, Forgot to remove this

Bod9001 committed:
Revert "oh, Forgot to remove this"

This reverts commit 0570cee1f2d04e0774690a79746468b6a74985d9.

Bod9001 committed:
ok, Let's not include unintentional stuff

Akbadain29 committed:
Modified Sprite Render

NoooneyDude committed:
Split PlayerFlippable from PlayerRotatable

NoooneyDude committed:
Merge pull request #5039 from Akbadain29/SpriteOrderVentScrubberFix

Modified Sprite Render

Aranclanos committed:
Refactors fuel consumption by the power generators.
Removes SolidPlasma.cs, all fuel consumption will now be handled by PowerGenerator.cs
Fixes power generators burning plasma instantly after the first sheet of the stack gets removed.
Players will now be allowed to add stacks to the generator.
Adds a player message warning for when the player tries to turn on a generator without power.
A plasma sheet will no longer be required for maps, the bool startAsOn will add a the corresponding fuel of one sheet to the generator at the start.

NoooneyDude commented on UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:
```diff
@@ -31,9 +29,10 @@ public class PowerGenerator : NetworkBehaviour, ICheckedInteractable<HandApply>,
-	private SolidPlasma burningSheet;
-	public bool IsFueled => itemSlot.IsOccupied || burningSheet != null;
+	private bool isOn;
+	private bool burningPlasma;
```
`burningPlasma` is never used.

NoooneyDude commented on UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs:
```diff
@@ -8,16 +8,14 @@ public class PowerGenerator : NetworkBehaviour, ICheckedInteractable<HandApply>,
 	[Tooltip("Whether this generator should start running when spawned.")]
 	[SerializeField]
-	private bool startAsOn = false;
+	private bool startAsOn;
```
Produces a compiler warning. https://i.imgur.com/4YWm1cO.png.

corp-0 replied:
yeah, you need to give it a default value

Akbadain29 committed:
Now warnText will be hiden when the UI is set for Joining

ReiderJack committed:
Merge pull request #5038 from NoooneyDude/fixes

Canister touchup

ReiderJack committed:
Merge pull request #5040 from Akbadain29/WarnTextAddition

Now warnText will be hidden when the UI is set for Joining

Bod9001 committed:
Merge pull request #5037 from Bod9001/Canister-fixes

Adds test for Objects that get set at 0,0  And removed some stuff that was causing warnings

ReiderJack committed:
Disallow removal of internal mags.

ReiderJack committed:
Fix Cigarette coroutine on Despawn.

ReiderJack committed:
Fix Cigarette starting coroutine on disabled object.

ReiderJack committed:
Remove pog from rotation

Bod9001 committed:
Removed a bunch of unneeded stuff

Bod9001 committed:
fixes some incorrect settings

AutumnNova committed:
explicitly check for maginternal as the allowMag checks are not enough

Bod9001 committed:
Merge pull request #5043 from the-other-autumn/smolgunfix

explicitly check for maginternal as the allowMag checks are not enough

Bod9001 committed:
Fixing stufff I removed accidentally and balances boiler

Bod9001 committed:
Merge pull request #5042 from Bod9001/ScriptClean

Script clean

Bod9001 committed:
Merge pull request #5044 from Bod9001/Reactor-power-Fix

Fixing stufff I removed accidentally and balances boiler

Aranclanos committed:
Merge branch 'generatorsheetmessage' of https://github.com/Aranclanos/unitystation into generatorsheetmessage
